api/all/src/main/java/io/opentelemetry/api/common/Value.java

```diff
@@ -84,6 +87,76 @@ static Value<List<KeyValue>> of(Map<String, Value<?>> value) {
     return KeyValueList.createFromMap(value);
   }
 
+  /**
+   * Convert a generic object to {@link Value}.
+   *
+   * <p>The following types are supported. If the {@code object} (or any nested data structures)
+   * contains an unsupported type, an {@link IllegalArgumentException} is thrown.
+   *
+   * <ul>
+   *   <li>{@link Long}
+   *   <li>{@link Integer}
+   *   <li>{@link Float}
+   *   <li>{@link Double}
+   *   <li>{@link Boolean}
+   *   <li>{@code byte[]}
+   *   <li>{@code List<?>}, where each list entry is a supported type
+   *   <li>{@code Map<String, ?>}, where each value is a supported type
+   * </ul>
+   *
+   * @param object the object to convert
+   * @return the equivalent {@link Value}
+   * @throws IllegalArgumentException if not able to convert the object to {@link Value}
```

Comment:
Might want to throw a checked exception to more forcibly signal to callers that they need to handle this.

Reply:
Decided against this:
- This method is analagous to jackson [ObjectMapper#convertValue](https://github.com/joansmith/jackson-databind/blob/230a19d527df7dc0c05017429bcb7aa1cbbe41ef/src/main/java/com/fasterxml/jackson/databind/ObjectMapper.java#L3421-L3426), which only throws a runtime exception. This API is quite popular and I've used it myself without issue for years, suggesting that a runtime exception is sufficient.
- I'm not sure what checked exception would be most appropriate to throw. Probably some subclass of `IOException`, and we'd probably want to introduce our own dedicated exception. In contrast, `IllegalArgumentException` pretty perfectly describes the error and seems appropriate to throw.